### ISSUE TITLE
Made a Frontend for repo using GitHub API, React and Chakra UI.

### ISSUE THREAD

**Issue by asheeeshh** [cat-milk/Anime-Girls-Holding-Programming-Books#358]
I'm new to frontend dev, so don't mind the design, made it in around 8 hours. Would definitely improve the UI, contributions are most welcome :)

Please let me know where can I improve.
![image](https://user-images.githubusercontent.com/68690233/156751723-b4d48cc9-9bfc-4ff3-96f5-7a47f6fe523d.png)

**Comment by asheeeshh** [comment]
Update: I worked on the UI a bit more and now the app has a separate landing page, and some UI related bugs have been fixed.

**Home**
![image](https://user-images.githubusercontent.com/68690233/156878808-118ba02d-6a88-422e-b709-0cf63903bd5b.png)

**App**
![image](https://user-images.githubusercontent.com/68690233/156878815-23427a54-72f6-4068-a89e-f40ef5deff48.png)

**Comment by asheeeshh** [comment]
**Todos:**
- Make Images popup on click (using modals).
- Add something like search bar, to let users search a particular anime girl after choosing a language.

I won't be too active on this project but I'll try to keep working, I have some exams next month so need to study 😔

**Comment by kotx** [comment]
Looks great! I found a few bugs though:
1. On dark mode browsers, the text on the landing page is black
![image](https://user-images.githubusercontent.com/33439542/156936794-b52d1365-2b9a-461d-aaea-e8ff6bea4bef.png)
2. The scrollbars and the image gallery is a bit wonky in terms of sizing, placement, etc:
![image](https://user-images.githubusercontent.com/33439542/156936786-2e0a861e-b652-44e6-8c4a-d38c9d3ffbca.png)
3. The page is initially empty, which can be a bit confusing- there should be a default selection to display all images, or have a placeholder that tells the user to pick a language.
![image](https://user-images.githubusercontent.com/33439542/156936835-b93f0cba-9a69-45c3-8ea0-7feecab29571.png)

**Comment by asheeeshh** [comment]
Thanks for letting me know @kotx :)

I'm unsure about why the first bug is happening, but maybe it's because of chakra ui's global styles, I'll try to fix it today only.

About scrollbars, I'm planning to style them something like this:
![image](https://user-images.githubusercontent.com/68690233/156953508-f53043db-64d2-4779-b137-8a1082d28e00.png)
Let me know if it would be better than the default one.

About the third one, I was thinking of this even myself, I'm thinking of just making it load images of a particular language by default, will this work?

**Comment by asheeeshh** [comment]
@cat-milk 
I was thinking that we can make an organization too and it will have all the web viewer(s), api(s) and api wrappers and the main repo of course. It will also remove the necessity to merge the project with this one, people can just ask you and add their projects to the organization.

**Comment by SoftwareGuy** [comment]
The question I would ask is would it be worth the hassle/effort though? When you say "add their projects", can you please elaborate?

**Comment by kotx** [comment]
Scrollbar looks good :+1:

**Comment by asheeeshh** [comment]
> The question I would ask is would it be worth the hassle/effort though? When you say "add their projects", can you please elaborate?

What I'm trying to say is having an organization will be much more cleaner and organized rather than trying to merge them in the main repo or keeping them separate, Not so sure if this is a good idea, but can you explain what do you mean by "worth the hassle/effort"?

**Comment by SoftwareGuy** [comment]
What I meant by my comment is what your proposing worth it in the long run?
How would it be cleaner/organized?

**Comment by kotx** [comment]
I don't see a significant difference in people owning their own repos as opposed to some org owning it.

**Comment by asheeeshh** [comment]
> What I meant by my comment is what your proposing worth it in the long run? How would it be cleaner/organized?

people would be able to find all the projects related to this repo in a single place? 🤔

**Comment by kotx** [comment]
What's wrong with a list of them in the README?

**Comment by asheeeshh** [comment]
> What's wrong with a list of them in the README?

nothing's wrong with it, this could work too.

**Comment by asheeeshh** [comment]
@kotx 
I think this is fixed now, can you check again?
> ![image](https://user-images.githubusercontent.com/68690233/157013468-06f3d8ba-8e3f-4159-8ab1-50836c6384b6.png)

**Comment by asheeeshh** [comment]
So I guess I have fixed almost everything now,

***List of changes I made:***
- Probably fixed the default browser theme bug.
- Changed the scrollbar style, it looks good now.
- On opening the app, it shows the images of **Rust Language** by default.
- Images popup on click, containing image name, and an option to download the image. (Thanks to [this amazing library](https://github.com/aautio/react-modal-image))

![image](https://user-images.githubusercontent.com/68690233/157034363-79410924-f5a0-46ce-8506-cdd638a14833.png)
![image](https://user-images.githubusercontent.com/68690233/157034392-c31ca4e4-d22c-4b35-9f0c-ef24ff5781c4.png)

**Comment by kotx** [comment]
> @kotx I think this is fixed now, can you check again?

Yup, looks much better, thanks!

Sorry to bug you again, but the image viewer's icons don't seem to be aligned with the image title and slightly offscreen:
![image](https://user-images.githubusercontent.com/33439542/157168402-14d8eb43-7e8d-42c8-9efd-89e1a3a51b92.png)
For some reason there's two scrollbars though. 
![image](https://user-images.githubusercontent.com/33439542/157168485-e2e42ef3-73d9-459c-8a79-19dc9f20fae7.png)
The images also have weird spaces between them when the size difference is too big.

Overall looks good though, thanks for making this!

**Comment by asheeeshh** [comment]
> > @kotx I think this is fixed now, can you check again?
> 
> Yup, looks much better, thanks!
> 
> Sorry to bug you again, but the image viewer's icons don't seem to be aligned with the image title and slightly offscreen: ![image](https://user-images.githubusercontent.com/33439542/157168402-14d8eb43-7e8d-42c8-9efd-89e1a3a51b92.png) For some reason there's two scrollbars though. ![image](https://user-images.githubusercontent.com/33439542/157168485-e2e42ef3-73d9-459c-8a79-19dc9f20fae7.png) The images also have weird spaces between them when the size difference is too big.
> 
> Overall looks good though, thanks for making this!

Yes, their are 2 scrollbars actually, one is for body and another is for the grid, because the grid was overflowing without adding the scroll.
About weird spacing between images, I can actually solve it by wrapping them inside divs but the thing is all the images have different sizes so forcing them to have the same resolution may cause them to get distorted, it wouldn't look good ig.
I'll just check if I can fix the icon alignment and let you know.

Thanks :)

**Comment by asheeeshh** [comment]
@kotx 
Icon alignment has been fixed now: asheeeshh/waifus-for-programmers@940bff1f5ae77e4bacd91147c96d8c4ac5adb783

**Comment by asheeeshh** [comment]
The UI is finally done now, I added the search bar and it's working quite fine, please let me know if you come across any bug :)

![image](https://user-images.githubusercontent.com/68690233/157239672-3e6afbfb-379f-41bb-9ead-169115d9a55d.png)

**Comment by kevinfengcs88** [comment]
This is epic.

**Comment by Asday** [comment]
> React is shit lol

Oh dude you fuckin' gottem bro.

**Comment by asheeeshh** [comment]
> React is shit lol

better than using vanilla js for me

**Comment by ccrsxx** [comment]
@marshallovski Tf you're talking about, vanilla is trash if you're building a website, you'll end up making your own shitty javascript framework if you use vanilla lol.

**Comment by asrvd** [comment]
update: rewrote the whole app using nextjs (much faster and better)

![image](https://user-images.githubusercontent.com/68690233/182364956-57df154b-49b6-4b19-9eaf-ed73c8209eb8.png)

(getting a better shorter domain for it soon)

**Comment by sonigeez** [comment]
LGTM

**Comment by all123all** [comment]
> React is shit lol

React nowadays is pretty fast, you can make it as faster as vanilla Javascript most of the time. Not only React, pretty much all of the modern frameworks such as vue and svelte are pretty fast as well. I really don't see why people hate React so much.
